test/functional/interface_zmq.py

```diff
@@ -43,66 +42,62 @@ def skip_test_if_missing_module(self):
         self.skip_if_no_py3_zmq()
         self.skip_if_no_bitcoind_zmq()
 
-    def setup_nodes(self):
+    def run_test(self):
         import zmq
+        self.ctx = zmq.Context()
+        try:
+            self.test_basic()
+            self.test_reorg()
+        finally:
+            # Destroy the ZMQ context.
+            self.log.debug("Destroying ZMQ context")
+            self.ctx.destroy(linger=None)
 
-        # Initialize ZMQ context and socket.
+    def test_basic(self):
         # All messages are received in the same socket which means
         # that this test fails if the publishing order changes.
         # Note that the publishing order is not defined in the documentation and
         # is subject to change.
-        self.zmq_context = zmq.Context()
-        socket = self.zmq_context.socket(zmq.SUB)
+        import zmq
```

Comment:
Is it normal to import stuff in the middle of a file in python? Don't see it often. And you could remove the one below in `test_reorg` if you just moved this to top..

Reply:
This was already the case.  It's like this to not break if zmq module is not available. See `skip_if_no_py3_zmq()` above.

Reply:
Ah, that makes sense..

Reply:
Can move the `import zmq` right after the line `skip_if_no_bitcoind_zmq`? :)

Reply:
Really? It's then available to other scopes? I'll try.

Reply:
If I remove import `zmq` and keep the one you said I get `NameError: name 'zmq' is not defined`.